examples/universal-cnf/README.md

```diff
@@ -1,7 +1,126 @@
+Universal CNF (UCNF) is an attempt to implement a swiss-army knife container, which implements basic networking functions useful for demoing NSM. It is currently heavily based on [VPP](https://github.com/FDio/vpp) and [vppagent](https://github.com/ligato/vpp-agent).
 
 ## Configuration
 
-TBD
\ No newline at end of file
+UCNF can be deployed and configured to implement various cloud-native netowrking scenarios. For this, it exposes a configuration file interface, where the solution integrator can program it.
+
+### The configuration file
+The configuration of UCNF is a yaml file that resides in `/etc/universal-cnf/config.yaml`. One would typically want to manage these configurations through Kubernetes ConfigMap and mount it as volume. Here is an incomplete `Deployment` which illustrates this concept:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  template:
+    spec:
+      containers:
+        - name: universal-cnf-client
+          image: networkservicemesh/universal-cnf-vppagent:latest
+          volumeMounts:
+            - mountPath: /etc/universal-cnf/config.yaml
+              subPath: config.yaml
+              name: universal-cnf-config-volume
+      volumes:
+        - name: universal-cnf-config-volume
+          configMap:
+            name: universal-cnf-client
+metadata:
+  name: ucnf-client
+```
+Then and sample `ConfigMap` can be modelled as follows:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: universal-cnf-client
+data:
+  config.yaml: |
+    initactions:
+      - command:
+          name: "vppctl"
+          args: ["show", "version"]
+```
+
+### Configuration concepts
+
+UCNF is implemented following the Network Service Mesh (NSM) networking model, where there are point to point links established for each service consumption request. Each Network Service (NS) is composed of Endpoints. The composition is happening when the NS Clients request a particular NS. These requests might or might not be labelled and the NSM distributed infrastructure consults the pre-defined `NetworkService` descriptors where the routing rules are defined. A typical CNF will expose one or more NS through one or more Endpoints and will connect to other services and/or Endpoints.
+
+UCNF configuration is based on events and actions. The supported events are generated on initialization and upon Endpoint connection request. The available actions are: command line execution, Client and dataplane configuration.
+
+Here is a more detailed explanation of the supported event types:
+ * The initialization event is the first event generated when UCNF starts. The actions associated with it will be executed before anything else in the system. This can typically be used to execute any pre-configured Clients or container wide commands.
+
+ * The Endpoint connection request event is generated when a Client requests a connection from the UCNF announced Endpoint.
+
+The actions which UCNF supports are as follows:
```

Comment:
Is the word here "actions" or it's by mistake? Can be a bit confusing and not clear how commands, client and dataplane are associated with actions.